### ISSUE TITLE
the issue when I change the batch size

### ISSUE THREAD

**Issue by r2zheng** [Henry1iu/TNT-Trajectory-Predition#1]
Hi,
I train the TNT with the original dataset from Argoverse and try to evaluate the accuracy of the model. But when I change the batch size in test_tnt.py from the default value 128 to 64, the evaluated minADE and minFDE are changed (using the same best_TNT and the same val_intermediate).
The trained model and the results are [here](https://drive.google.com/drive/folders/1ggr8yA-dBjBqYqJdu-6n4PECKIgsgxNV?usp=sharing).
Is there any reason for this?
Thank you!

**Comment by 12num** [comment]
你好，我最近也在跑这个程序。在train_vectornet.py中训练出dataset/interm_data/train_intermediate/processed/data.pt的文件时 因为内存大小限制失败，请问您的内存大小是多少呢？如果可以的话，分享给我您的data.pt文件可以吗

**Comment by Henry1iu** [comment]
> Hi, I train the TNT with the original dataset from Argoverse and try to evaluate the accuracy of the model. But when I change the batch size in test_tnt.py from the default value 128 to 64, the evaluated minADE and minFDE are changed (using the same best_TNT and the same val_intermediate). The trained model and the results are [here](https://drive.google.com/drive/folders/1ggr8yA-dBjBqYqJdu-6n4PECKIgsgxNV?usp=sharing). Is there any reason for this? Thank you!

Thank you for raising the first issue!!!

This issue is caused by the normalization operation within the subgraph of the Vectornet backbone. Actually, I have changed the implementation of Vectornet Backbone. (Because I found the pytorch_geometric is not necessary for the Vectornet implementation.) For the latest version, you can refer to the version in the branch called jb. I will merge my modification to the main branch later. Currently, I'm still checking if the modified version has the same issue. I'll notify you once the results come out.

**Comment by Henry1iu** [comment]
> 你好，我最近也在跑这个程序。在train_vectornet.py中训练出dataset/interm_data/train_intermediate/processed/data.pt的文件时 因为内存大小限制失败，请问您的内存大小是多少呢？如果可以的话，分享给我您的data.pt文件可以吗

内存至少需要128G, 目前我的设备设置的是: memory 64G, Swap 128G
data.pt 文件因为容量比较大,分享起来比较麻烦

**Comment by r2zheng** [comment]
Hi sorry for the late response. I uploaded the data.pt to this [link](https://pan.baidu.com/s/1MdDPhTB9QLKGmb4vhXMXOw), and the password is 1234. Due to some limitations, I split the file into five pieces, so please use `cat data.pt_split.?? > data.pt` to join them. By the way, I used train_tnt.py instead of train_vectornet.py, and I don't think the problem is memory since the memory on my device is 256GB (and the peak during the training is approximately 140G).
我把data.pt上传到上面的链接了，密码是1234，下载后麻烦合并一下。我设备的内存上限是256G，训练时的上限大约是140G，并且我训练的模型是TNT。不过感谢你们的解答，如果有什么解决方案或还有什么信息需要提供请联系我，谢谢。

**Comment by Henry1iu** [comment]
Hi, I have fixed the issue with changing the batch size. The normalization in the subgraph implementation causes this issue. I changed the normalizing dimension and pushed my new subgraph implementation. Surprisingly, the performance of VectorNet also improves a lot. I'm still testing the performance of TNT. You can give this new commit a try if you want!
